src/pyrtma/client_logging.py

```diff
@@ -0,0 +1,369 @@
+import logging
+import logging.handlers
+import pyrtma.core_defs as cd
+import pathlib
+
+from .message import MessageData
+from .exceptions import ClientError
+
+from typing import Union, Type, Dict, Optional, Protocol
+
+
+class ClientLike(Protocol):
```

Comment:
I'm not really familiar with `typing.Protocol`, but would it be worth making an abstract base class that both `Manager` and `Client` inherit from?

Reply:
@dmweir `typing.Protocol` was added in python 3.8. I think we want to drop support for python 3.7, but we likely still have python 3.7 environments on the older rigs. I think our options are either to review and update conda installations on all of our machines asap (including home rigs, surface laptops, etc.), or to try to maintain 3.7-compatibility for now.

Reply:
I used typing.Protocol to avoid circular import issues that I ran into. An abstract base may work if we can avoid that issue.

Reply:
I think I was wrong about github dropping 3.7 support, that may have been a decision by us in the past. 3.7-3.12 should be preinstalled on the runners. Other versions are possible to test with extra work, but adding 3.7 and/or 3.12 should just require changing one line in the workflow file.

Reply:
Changed from typing.Protocol to ABC in 1873bdb4.